lib/services/elasticsearch.js

```diff
@@ -679,17 +679,24 @@ class ElasticSearch extends Service {
         // If some errors occured during the Bulk, we send a "Partial Error" response :
         if (result.errors) {
           result.partialErrors = [];
+          const items = [];
+          let row;
 
-          for (const resultItem of Object.keys(result.items)) {
-            for (const action of Object.keys(result.items[resultItem])) {
-              const item = result.items[resultItem][action];
+          while ((row = result.items.shift()) !== undefined) {
+            const
+              action = Object.keys(row)[0],
+              item = row[action];
 
-              if (item.error) {
-                item.action = action;
-                result.partialErrors.push(item);
-              }
+            if (item.error) {
+              item.action = action;
+              result.partialErrors.push(item);
+            }
+            else {
+              items.push(item);
             }
           }
+
+          result.items = items;
```

Comment:
boyscouting: split valid from errored items in result.items vs result.partialErrors